tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub get_test_name {
+    return my $name = (caller(2))[3];
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
+    my $description;
+    my $result;
+
+    for my $i (@results) {
+        if ($i eq 'test') {
+            $test_name = $test{$i};
+        } elsif ($i eq 'description') {
+            $description = $test{$i};
+        } elsif ($i eq 'result') {
+            if ($test{$i} eq 'FAIL') {
+                $result = $test{$i};
+            }
+        } else {
+            die('Wrong key!');
+        }
+    }
+
+    if ($result eq 'FAIL') {
+        script_run("echo \"<testcase name='$test_name' errors='1'>\" | tee -a $file");
+    } else {
+        script_run("echo \"<testcase name='$test_name'>\" | tee -a $file");
+    }
+    script_run("echo \"<system-out>\" | tee -a $file");
+    script_run("echo $description | tee -a $file");
+    script_run("echo \"</system-out>\" | tee -a $file");
+    script_run("echo \"</testcase>\" | tee -a $file");
+}
+
+sub test_01 {
+    my $description = 'Basic SRUN test';
+    my $result;
+
+    $result = script_run("srun -w slave-node00 date");
+
+    my %results = results($description, $result);
+    return %results;
```

Comment:
```suggestion
    return results($description, $result);
```

Reply:
I'm not sure I like such approach. I prefer return value to be explicit, so that one does not need to check what another function returns.
In any case if you insist this is better, I'll change it. Wdyt?

Reply:
keep in mind that now this function has self-speaking name `generate_result_hash` so your concerns should be less relevant , right ? But I would say other way around - if you insist you may leave it :) after all your version is not wrong just not apply to my and Jose's coding style :)

Reply:
I don't have hard feelings here.
My concerns would be, using hashref instead of associative array. As each assignment would create a copy. But everything is quite small here, so nothing I would worry for.